prototype/mobile_endpoint/case/case_processing.py

```diff
@@ -0,0 +1,239 @@
+from collections import namedtuple
+import copy
+from datetime import datetime
+import logging
+
+from mobile_endpoint.case import const
+from mobile_endpoint.case.models import CommCareCase
+from mobile_endpoint.case.xml.parser import case_update_from_block
+from mobile_endpoint.dao import CaseDbCache
+from mobile_endpoint.exceptions import ReconciliationError
+from mobile_endpoint.form.models import XFormInstance
+
+
+logger = logging.getLogger(__name__)
+
+
+def process_cases_in_form(xform, dao):
+    with CaseDbCache(dao, domain=xform['domain'],
+                     lock=True, deleted_ok=True, xforms=[xform]) as case_db:
+        case_result = _get_or_update_cases([xform], case_db)
+        cases = case_result.cases
+
+        for case in cases:
+            case['server_modified_on'] = datetime.utcnow()
+            case_db.mark_changed(case)
+            if not case.check_action_order():
+                try:
+                    case.reconcile_actions(rebuild=True, xforms={xform.id: xform})
+                except ReconciliationError:
+                    pass
+
+        # todo: check that cases haven't been modified since we loaded them (why not use locking?)
+        # todo: update synclog
+
+        return case_db.get_changed()
+
+
+def _get_or_update_cases(xforms, case_db):
+    """
+    Given an xform document, update any case blocks found within it,
+    returning a dictionary mapping the case ids affected to the
+    couch case document objects
+    """
+    # have to apply the deprecations before the updates
```

Comment:
I should have deleted the comment. There was a line below that I deleted (but could also put back):
```
sorted_forms = sorted(xforms, key=lambda f: 0 if is_deprecation(f) else 1)
```